### ISSUE TITLE
[BUG] Cookies not set in headless mode

### ISSUE THREAD

**Issue by rockwalrus** [microsoft/playwright#12884]
**Context:**
- Playwright Version: 1.18.1
- Operating System: Windows 10
- Node.js version: 16.13.0
- Browser: Chromium
- Extra: Chromium is run with `--no-sandbox`

**Describe the bug**

When run normally, the browser is accepting the `Set-Cookie` header. However, when run with `show` set to false (i.e. headless mode), the browser ignores the `Set-Cookie` header.

This bug differs from its tens of near-duplicates because:
 * It affects first-party as well as third-party cookies.
 * It's verified to only happen in headless mode.
 * It's not because of headless detection. Packet interception shows that the `Set-Cookie` header is being sent by the server.
 
**Workaround**
The magical incantation `--disable-features=ImprovedCookieControls,ImprovedCookieControlsForThirdPartyCookieBlocking,SameSiteByDefaultCookies,CookiesWithoutSameSiteMustBeSecure` makes the symptoms go away, at least for right now.

**Comment by pavelfeldman** [comment]
Could you share a repro scenario? Our regular tests for this work, so there must be something special about your use case.

**Comment by dgozman** [comment]
We need more information to act on this report. Please file a new one and link to this issue when you get back to it!

**Comment by develorer** [comment]
I am facing the exact same issue on python.

I am launching a persistent context with user-data-dir:

`sync_playwright.chromium.launch_persistent_context`

If I set `headless=False`, everything goes well, cookies are being picked up correctly from the user-data-dir.

If I set `headless=True`, the browser starts with no cookies.

I tried the "magical incantation" of @rockwalrus , but it didn't work. Any chance we could re-open this? And does anyone know a workaround for the short term?
